pkg/apis/core/validation/shoot.go

```diff
@@ -2070,6 +2076,20 @@ func validateCoreDNS(coreDNS *core.CoreDNS, fldPath *field.Path) field.ErrorList
 	return allErrs
 }
 
+// ValidateFinalizersOnCreation validates the finalizers of a Shoot object.
+func ValidateFinalizersOnCreation(finalizers []string, fldPath *field.Path) field.ErrorList {
+	allErrs := field.ErrorList{}
+
+	for i, finalizer := range finalizers {
+		idxPath := fldPath.Index(i)
+		if ForbiddenShootFinalizersOnCreation.Has(finalizer) {
+			allErrs = append(allErrs, field.Forbidden(idxPath, fmt.Sprintf("finalizer %q cannot be added on creation", finalizer)))
```

Comment:
Nit, not necessarily required.
```suggestion
		if ForbiddenShootFinalizersOnCreation.Has(finalizer) {
			allErrs = append(allErrs, field.Forbidden(fldPath.Index(i), fmt.Sprintf("finalizer %q cannot be added on creation", finalizer)))
```